<launch>
    <arg name="device_ip" default="192.168.1.200" />
    <arg name="msop_port" default="2368" />
    <arg name="difop_port" default="2369" />
    <arg name="lidar_type" default="ch120"/>
    <arg name="pcl_type" default="false"/>
    <arg name="use_time_service" default="false" />

  <node pkg="lslidar_driver" type="lslidar_ch_driver_node" name="lslidar_driver_node" output="screen">
    <!--param name="pcap" value="/home/ls/Downloads/ch120/ros.pcap"/-->
    <param name="lidar_ip" value="$(arg device_ip)"/>
    <param name="lidar_type" value="$(arg lidar_type)"/>
    <param name="msop_port" value="$(arg msop_port)" />
    <param name="difop_port" value="$(arg difop_port)"/>
    <param name="pcl_type" value="$(arg pcl_type)"/>
    <param name="add_multicast" value="false"/>
    <param name="group_ip" value="224.1.1.2"/>
    <param name="frame_id" value="laser_link"/>
    <param name="min_range" value="0.15"/>
    <param name="max_range" value="500.0"/>
    <param name="packet_rate" value="2525.0"/>
    <param name="angle_disable_min" value="0"/>
    <param name="angle_disable_max" value="0"/>
    <Param name="pointcloud_topic" value="lslidar_point_cloud"/>
    <param name="horizontal_angle_resolution" value="0.36"/>   <!-- 10Hz:0.36  20Hz:0.72 5Hz: 0.18 -->
    <param name="use_time_service" value="$(arg use_time_service)"/>
    <param name="echo_num" value="0"/>
    <param name="publish_laserscan" value="false"/>
    <param name="channel_num" value="60"/>
  </node>
</launch>
